### PR TITLE
Support parse Enable command response

### PR DESCRIPTION
**Pull request by fansu**
## Description
Support parse Enable command response, add EnableResult class and its parser

## Motivation and Context
Support parse Enable command response for ENABLE command

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Major release (change is NOT backward compatible with prior release)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: core/pom.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
     <parent>
         <groupId>com.yahoo.imapnio</groupId>
         <artifactId>imapnio</artifactId>
-        <version>4.3.6</version>
+        <version>4.3.7</version>
```
Comment (jui8wang) [inline core/pom.xml:8]: fix readme too

File: core/src/test/java/com/yahoo/imapnio/async/response/ImapResponseMapperTest.java [hunk 2]
```diff
@@ -1299,4 +1301,123 @@ public void testParseToSearchResultNotOK() throws IOException, ProtocolException
...
+     * @throws IOException will not throw
+     * @throws ProtocolException will not throw
+     * @throws ImapAsyncClientException will not throw
+     */
+    @Test
+    public void testParseToEnableResult() throws IOException, ProtocolException, ImapAsyncClientException {
+        final ImapResponseMapper mapper = new ImapResponseMapper();
+        final List<IMAPResponse> content = new ArrayList<>();
+        content.add(new IMAPResponse("* some junks\r\n"));
+        content.add(new IMAPResponse("* ENABLED CONDSTORE\r\n"));
```
Comment (jui8wang) [inline core/src/test/java/com/yahoo/imapnio/async/response/ImapResponseMapperTest.java]: Make it one line 
* ENABLED CONDSTORE QRSYNC\r\n

**Review by jui8wang — CHANGES_REQUESTED** [review]

File: core/src/main/java/com/yahoo/imapnio/async/response/ImapResponseMapper.java [hunk 3]
```diff
@@ -467,6 +473,40 @@ private IdResult parseToIdResult(@Nonnull final IMAPResponse[] ir) throws ImapAs
...
+
+            final Set<String> enabledCapabilities = new HashSet<String>();
+            for (final IMAPResponse resp : ir) { // ex: CONDSTORE QRSYNC
+                if (resp.isTagged()) {
+                    continue;
+                }
+                if (resp.keyEquals("ENABLED")) {
+                    String s;
+                    while ((s = resp.readAtom()) != null && s.length() != 0) {
+                        final String c = s.replaceAll("(\\r|\\n)", ""); // ex: CONDSTORE
```
Comment (nvsnvikram) [inline core/src/main/java/com/yahoo/imapnio/async/response/ImapResponseMapper.java:501]: I am not sure if this is needed. Please check.

File: core/pom.xml (hunk 1 above)
Comment (fansu) [reply core/pom.xml:8]: ignore, as discussed

File: core/src/test/java/com/yahoo/imapnio/async/response/ImapResponseMapperTest.java [hunk 4]
```diff
@@ -1299,4 +1301,120 @@ public void testParseToSearchResultNotOK() throws IOException, ProtocolException
...
+
+    /**
+     * Tests parseToEnableResult method when server returns with not OK response.
+     *
+     * @throws IOException will not throw
+     * @throws ProtocolException will not throw
+     * @throws ImapAsyncClientException will not throw
+     */
+    @Test
+    public void testParseToEnableResultNotOK() throws IOException, ProtocolException, ImapAsyncClientException {
```
Comment (arun-prakash) [inline core/src/test/java/com/yahoo/imapnio/async/response/ImapResponseMapperTest.java:1404]: Need one more UT to handle the empty enabled response, 
sample
```
C: a1 ENABLE UNAVAILABLE-CAPABILITY
S: * ENABLED
S: a1 OK ENABLE completed
```

File: core/src/test/java/com/yahoo/imapnio/async/response/ImapResponseMapperTest.java (hunk 2 above)
Comment (fansu) [reply core/src/test/java/com/yahoo/imapnio/async/response/ImapResponseMapperTest.java]: done

**Comment by arun-prakash** [comment]
Looks good

**Review by nvsnvikram — APPROVED** [review]

**Pull request merged**